enzyme/Enzyme/MLIR/Dialect/EnzymeOps.td

```diff
@@ -51,4 +55,71 @@ def ForwardDiffOp : Enzyme_Op<"fwddiff",
   }];
 }
 
+def DiffOp : Enzyme_Op<"diff",
+    [DeclareOpInterfaceMethods<SymbolUserOpInterface>]> {
+  let summary = "Perform reverse mode AD on a funcop";
+  let arguments = (ins FlatSymbolRefAttr:$fn, Variadic<AnyType>:$inputs, ActivityArrayAttr:$activity);
+  let results = (outs Variadic<AnyType>:$outputs);
+
+  let assemblyFormat = [{
+    $fn `(` $inputs `)` attr-dict `:` functional-type($inputs, results)
+  }];
+}
+
+def PushCacheOp : Enzyme_Op<"push"> {
+  let summary = "Store information for the reverse mode pass.";
+  let arguments = (ins AnyType : $cache, AnyType : $value);
+  let results = (outs );
+}
+
+def PopCacheOp : Enzyme_Op<"pop"> {
+  let summary = "Retrieve information for the reverse mode pass.";
+  let arguments = (ins AnyType : $cache);
+  let results = (outs AnyType:$output);
+}
+
+def CreateCacheOp : Enzyme_Op<"init"> {
+  let summary = "Creat cache for reverse pass.";
```

Comment:
Could you clarify the use of the word "cache" here? It seems from my testing that this isn't creating a cache in the AD sense (storing required primal values), it's more used to allocate space for the gradient. Is it meant to be overloaded to do both?